File: ContinousCards/ContinousCards.cpp
class Solution {
public:
    bool IsContinuous( vector<int> nums ) {
        int n = nums.size();
        if (nums.empty()|| n!=5)
            return false;
        sort(nums.begin(),nums.end());
        int numZero = 0;
        int gap = 0;
        for (int i=0;i<n;i++)
        {
            if (nums[i]==0)
                numZero++;
        }
        for (int i=numZero, j=numZero+1;j<n;i++,j++)
        {
            if (nums[i]==nums[j]) return false;
            gap += nums[j]-nums[i]-1;
        }
        if (numZero<gap) return false;
        return true;
    }
};

File: ContinousCards/ContinousCards.py
# -*- coding:utf-8 -*-
class Solution:
    def IsContinuous(self, nums):
        # write code here
        n = len(nums)
        if nums is None or n!=5:
            return False
        nums = sorted(nums)
        numZero = 0
        gap = 0
        for num in nums:
            if num ==0:
                numZero+=1
        for i in range(numZero,n-1):
            j=i+1
            if nums[j]==nums[i]:
                return False
            gap += nums[j]-nums[i]-1
        if gap > numZero:
            return False
        return True
